From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 17 Jan 2022 17:05:18 +0800
Subject: [PATCH] net/txgbe: fix compile issue of txgbe_ethtool.c
Patch-mainline: Never, fixes out-of-tree driver
References: patch dependencies

Since commit 7462494408cd3 ("ethtool: extend ringparam setting/getting
API with rx_buf_len") addes two new parameters kernel_ringparam and
extack for .get_ringparam and .set_ringparam.

Let's update netswift txgbe driver to reflect the change.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/netswift/txgbe/txgbe_ethtool.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/netswift/txgbe/txgbe_ethtool.c b/drivers/net/ethernet/netswift/txgbe/txgbe_ethtool.c
index ca7265259a36..4b77654bacda 100644
--- a/drivers/net/ethernet/netswift/txgbe/txgbe_ethtool.c
+++ b/drivers/net/ethernet/netswift/txgbe/txgbe_ethtool.c
@@ -1050,7 +1050,9 @@ static void txgbe_get_drvinfo(struct net_device *netdev,
 }
 
 static void txgbe_get_ringparam(struct net_device *netdev,
-				struct ethtool_ringparam *ring)
+				struct ethtool_ringparam *ring,
+				struct kernel_ethtool_ringparam *kernel_ering,
+				struct netlink_ext_ack *extack)
 {
 	struct txgbe_adapter *adapter = netdev_priv(netdev);
 
@@ -1065,7 +1067,9 @@ static void txgbe_get_ringparam(struct net_device *netdev,
 }
 
 static int txgbe_set_ringparam(struct net_device *netdev,
-			       struct ethtool_ringparam *ring)
+			       struct ethtool_ringparam *ring,
+			       struct kernel_ethtool_ringparam *kernel_ring,
+			       struct netlink_ext_ack *extack)
 {
 	struct txgbe_adapter *adapter = netdev_priv(netdev);
 	struct txgbe_ring *temp_ring;
-- 
2.26.2

